install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -103,3 +103,67 @@ spec:
             severity: warning
           annotations:
             message: "Memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90%. Master nodes starved of memory could result in degraded performance of the control plane."
+    - name: control-plane-cpu-utilization
+      rules:
+        - alert: HighOverallControlPlaneCPU
+          annotations:
+            summary: >-
+              CPU utilization across all three control plane nodes is higher than two control plane nodes can sustain; a single control plane node outage may
+              cause a cascading failure; increase available CPU.
+            runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/machine-config-operator/ExtremelyHighIndividualControlPlaneCPU.md
+            description: >-
+              Given three control plane nodes, the overall CPU utilization may only be about 2/3 of all available capacity.
+              This is because if a single control plane node fails, the remaining two must handle the load of the cluster in order to be HA.
+              If the cluster is using more than 2/3 of all capacity, if one control plane node fails, the remaining two are likely to
```

Comment:
This assumes there are three control-plane nodes.  We have Telemetry metrics that tell us how many there are, if we wanted to craft something more flexible.  [This][1] is an example of launching queries when the alert matches in order to feed in data that's not included in the `expr`.  In this case, it would be something like:

```
{{ "{{ with $count_control_plane := \"count by (node) (kube_node_role{role=\"master\"})\" | query }}{{ if eq (first $count_control_plane) 1}} Blah, blah, single node...{{ howeverWeDoElse }}...
```

[1]: https://github.com/openshift/cluster-version-operator/blob/21f8f50e95d5237eb4d82f3e2ca4c47bacf74d3f/install/0000_90_cluster-version-operator_02_servicemonitor.yaml#L46

Reply:
Is it even meaningful to tell them how many control-plane nodes they have in the cluster? I would personally be in favor of making the sentence generic.

Reply:
In a 3-node control plane, keeping resources below 66% makes sense, because you'll have enough space to drain out a member while rolling the pool.  With a single-nide control plane, reserving 40% capacity just for handling load bursts seems... certainly less obvious to me, even if it is still warranted for some reason.  If you have clear reasoning for why folks should care about >60% consumption that does not rely on the number of control plane nodes, that sounds like a great approach to me, but I don't see that reasoning myself yet.